### PR TITLE
Run build with all supported python versions before publishing to pypi

### LINKED ISSUE #3577: Test different Python versions in Airbyte CDK
## Tell us about the problem you're trying to solve
In the Airbyte CDK CI we only test the package on Python 3.7: https://github.com/airbytehq/airbyte/blob/master/.github/workflows/publish-cdk-command.yml#L44

we should change the CI such that it tests the package on all Python versions supported in [setup.py](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/setup.py#L56). 

## Describe the solution you’d like
Whenever the `/publish-cdk` workflow is triggered, it should first test the connector on all python versions supported before publishing it to Pypi.

To do this, we should find a way to run the workflow coded in `.github/workflows/publish-cdk.yaml` using all the versions of python we claim to support. 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200376873960124) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by midavadim**
## What
Test build with different supported python versions

## How
Added strategy with list of supported python versions [3.7.1, 3.7, 3.8, 3.9]
where 3.7.1 - is a minimal of 3.7.X versions supported in github actions
So build is running in different python envs.

## Pre-merge Checklist
- not possible to run workflow without committing it into the master branch.

### CODE REVIEW

**Review by davinchia — COMMENTED** [review]
Looks good!

@midavadim did you do any manual test to double check this setup works fine? Want to quickly discuss that bit before we merge this in.

**Comment by midavadim** [comment]
@davinchia 
I've just made additional updates.

I tested it on my fork of airbyte. It looks fine (with exception for actual publishing task)

**Review by davinchia — APPROVED** [review]
@midavadim great. there was a transient build error. I just restarted the build. feel free to merge this in after the build succeeds.

have 1 question, but it's non-blocking.

File: .github/workflows/publish-cdk-command.yml
Comment (davinchia) [reply .github/workflows/publish-cdk-command.yml:19]: NVM, I realised it's the minimal of the 3.7.X.

Is there a reason we need to build the 3.7.X versions in additional to 3.7?

**Pull request merged**
